torch/csrc/distributed/c10d/CUDASymmetricMemory.cu

```diff
@@ -108,8 +112,15 @@ class IpcChannel {
 
  private:
   static std::string get_socket_name(int pid) {
+    const char* tmp_dir = "/tmp";
+    for (const char* env_var : {"TMPDIR", "TMP", "TEMP", "TEMPDIR"}) {
+      if (const char* path = getenv(env_var)) {
+        tmp_dir = path;
+        break;
+      }
+    }
```

Comment:
You should use std::filesystem::temp_directory_path is what should be used to get this properly in stdlib.

Reply:
Merged before I saw this. Addressed in https://github.com/pytorch/pytorch/pull/130842 (also replacing another occurrence).